packages/flutter/lib/src/widgets/will_pop_scope.dart

```diff
@@ -85,13 +85,19 @@ class _WillPopScopeState extends State<WillPopScope> {
   @override
   void didUpdateWidget(WillPopScope oldWidget) {
     super.didUpdateWidget(oldWidget);
-    assert(_route == ModalRoute.of(context));
-    if (widget.onWillPop != oldWidget.onWillPop && _route != null) {
+    final ModalRoute<dynamic>? actualRoute = ModalRoute.of(context);
```

Comment:
Checking the route should be done in didChangeDependencies

Reply:
@chunhtai can you please make review? and help with failed checks?